app/models/user.rb

```diff
@@ -6,9 +6,14 @@ class User < ApplicationRecord
          :recoverable, :rememberable, :validatable,
          :omniauthable, omniauth_providers: %i[facebook]
 
-  has_many :friendships, foreign_key: :user_id, class_name: 'Friendship'
+  has_many :friendships, foreign_key: :friend_sender_id, class_name: 'Friendship'
```

Comment:
I am not sure why you changed the name, this does not change how it works


Check this article out https://smartfunnycool.com/friendships-in-activerecord/

Reply:
It's not just a name change its a join I left a note in my last commit. User that sends a friend requests their id will be populated in the friend_sender association of the user that received the request. The user that the friend_request was sent too will be generated